scripts/local-registry/populate-storage.js

```diff
@@ -0,0 +1,56 @@
+// @ts-check
+
+const { startLocalRegistry } = require('@nx/js/plugins/jest/local-registry');
+const { exec } = require('node:child_process');
+const {
+  LARGE_BUFFER,
+} = require('nx/src/executors/run-commands/run-commands.impl');
+
+async function populateLocalRegistryStorage() {
+  try {
+    const publishVersion = process.env.PUBLISHED_VERSION ?? 'major';
+    const isVerbose = process.env.NX_VERBOSE_LOGGING === 'true';
+    const registryTeardown = await startLocalRegistry({
+      localRegistryTarget: '@nx/nx-source:local-registry',
+      verbose: isVerbose,
+      clearStorage: true,
+    });
+
+    console.log('Publishing packages to local registry to populate storage');
+    await runLocalRelease(publishVersion, isVerbose);
+
+    registryTeardown();
```

Comment:
You can do this as a followup since this is an existing issue but could you please ensure that when publishing fails, that `registryTeardown()` is called please? And I guess we have to fix it in the version which does the local release as well.

Reply:
https://github.com/nrwl/nx/pull/23470